tests/drivers/counter/counter_basic_api/src/test_counter.c

```diff
@@ -147,22 +147,22 @@ static void top_handler(struct device *dev, void *user_data)
 {
 	zassert_true(user_data == exp_user_data,
 			"%s: Unexpected callback", dev->config->name);
-	top_cnt++;
+	k_sem_give(&top_cnt_sem);
 }
 
 void test_set_top_value_with_alarm_instance(const char *dev_name)
 {
 	struct device *dev;
 	int err;
 	u32_t cnt;
-	u32_t tmp_top_cnt;
+	u32_t top_cnt;
 	struct counter_top_cfg top_cfg = {
 		.callback = top_handler,
```

Comment:
Setting top value with this configuration in user mode should fail. Am I right?

Reply:
Correct. That's why `test_set_top_value_with_alarm()` is not run in user mode.

Reply:
right, as i said: not very familiar with user mode.

Reply:
@nordic-krch Sorry about the confusion. I had pushed a half-finished solution. The correct solution is pushed now.